README.md

```diff
@@ -1,9 +1,39 @@
-over named pipes in Windows. A container, such as CSI node plugins, can mount
-the named pipes depending on operations it wants to exercise on the host and
-invoke the APIs.
+CSI Proxy (which might be more aptly named "csi-node-proxy") is a binary that exposes a set of gRPC APIs around local storage operations for nodes in Windows. A container, such as CSI node plugins, can mount the named pipes depending on operations it wants to exercise on the host and
+invoke the APIs.  This allows a storage plugin to run as if were a CSI plugin on linux which had the ability to do "priveleged" actions on the windows host OS.
+
+```
+                                              +------------------+
++---------------------------------------------|                  |
+|                 +----------+                |                  |
+|                 |csi node  |                |   CSI Node Plugin|
+|                 |driver    +---------------->                  |
+|                 |registrar |                |                  |
+| NodeStageVolume ----^------+                ----------+--------+
+| NodeUnstageVol      |                                 |  Disk APIs
+| NodePublishVol      |GetNodeInfo                      |  Volume APIs
+| NodeResizeVol       |NodeRegistrationStatus    /pipe  |  SMB APIs
+| NodeGetInfo         |                                 |  FileSystem/System APIs
+| NodeGetVolStats     | csi-plugin-reg.sock   +---------v---------+
+|                     |                       |                   |
+|                 + --+--------+              |    csi-proxy.exe  |
+|                 | kubelet.exe|              +-------------------+
+|                 |            |                      |
+|                 +---+--------++                     |
+|                     |                               |
+| csi.sock            |                               |
+|                     v                               v
+|                 +---+-------------------------------+------+
+|                 |                                          |
++---------------->|         Host Os (Windows)                |
+                  +------------------------------------------++
+```
+In the above diagram, there are 3 communication channels being utilized for CSI proxy:
+
+- `csi.sock` : This lives at /var/lib/kubelet/plugins/csi-plugin/csi.sock, and is used for the host OS to communicate with the CSI node plugin
+- `csi-plugin-reg.sock` : This lives here, /var/lib/kubelet/plugins_registry/csi-plugin-reg.sock, and is used for the kubelet itself to get metadata as needed from the csi plugin  
+- `/pipe` : /pipe/csi-proxy-api-group-v-... : This pipe lives at \pipe\csi-proxy-api-group-v-..., upon receipt of specific requests from the Node plugin, it carries out actions on the node plugins behalf.  Since it doesn't execute arbitrary commands on the host, its more secure then simply being a black-box proxy to run host commands.
```

Comment:
Please use `\\.\pipe\csi-proxy-...` for named pipe path prefix.

Reply:
ok will fix this shortly.